File: src/ios/Progress.m
//
//  Progress.m
//  Copyright (c) 2015 Lee Crossley - http://ilee.co.uk
//

#import "Cordova/CDV.h"
#import "Cordova/CDVViewController.h"
#import "Progress.h"
#import "KVNProgress.h"

@implementation Progress

- (void) show:(CDVInvokedUrlCommand*)command;
{
    NSMutableDictionary *args = [command.arguments objectAtIndex:0];

    [KVNProgress showWithStatus:[args objectForKey:@"text"]];

    [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK] callbackId:command.callbackId];
}

- (void) update:(CDVInvokedUrlCommand*)command;
{
    NSMutableDictionary *args = [command.arguments objectAtIndex:0];

    [KVNProgress updateStatus:[args objectForKey:@"text"]];

    [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK] callbackId:command.callbackId];
}

- (void) hide:(CDVInvokedUrlCommand*)command;
{
    [KVNProgress dismissWithCompletion:^{
        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK] callbackId:command.callbackId];
    }];
}

@end

File: www/progress.js

var exec = require("cordova/exec");

var Progress = function () {
    this.name = "Progress";
};

Progress.prototype.show = function (onSuccess, onError, options) {
    if (arguments.length === 1 && typeof (arguments[0]) !== "function") {
        options = arguments[0];
        arguments[0] = null;
    }
    if (typeof (options) === "string") {
        options = {"text": options};
    }
    if (typeof (options) === "undefined") {
        options = {"text": ""};
    }
    exec(onSuccess, onError, "Progress", "show", [options]);
};

Progress.prototype.update = function (onSuccess, onError, options) {
    if (arguments.length === 1 && typeof (arguments[0]) !== "function") {
        options = arguments[0];
        arguments[0] = null;
    }
    if (typeof (options) === "string") {
        options = {"text": options};
    }
    if (typeof (options) === "undefined") {
        options = {"text": ""};
    }
    exec(onSuccess, onError, "Progress", "update", [options]);
};

Progress.prototype.hide = function (onSuccess, onError) {
    exec(onSuccess, onError, "Progress", "hide");
};

module.exports = new Progress();
